Replace native clients

agocke opened the pull request:
Same pull request as before, this one just bases against my previous branch as opposed to master to avoid duplicate commits and comments. Following is the text from the original PR:

This change continues on the trend of removing the native client code for the compiler server and replacing it with managed code. This is the PR which actually removes the native code! :) This commit deletes all of the native projects and replaces them with the managed client implementation, keeping exactly the same functionality as the old csc2/vbc2 projects.

As before, this PR builds on the commits of the previous one, namely #866, so that one should be reviewed and merged first.

@gafter @jaredpar @AlekseyTs @VladimirReshetnikov @VSadov @pharring

jaredpar commented on BuildAndTest.proj:
```diff
@@ -75,10 +69,9 @@
-      ShadowCopy="false" />
+    <exec
+      Command="packages\xunit.runners.2.0.0-rc1-build2826\tools\xunit.console.exe @(TestAssemblies, ' ') -noshadow -parallel assemblies" />
+
```
Do we have an issue to remove the assemblies argument to parallel?  Is it worth doing?

agocke replied:
I think we would need an issue just to investigate that, but I suspect we want to parallelize more. My CPU spends a fair amount of time idle when running the tests.

jaredpar commented:
This is currently a PR into your own fork.  Did you mean to do a PR into dotnet/roslyn instead?

agocke commented:
Unfortunately, this repo is a fork of your old git repo, so it doesn't recognize it as a fork of dotnet/roslyn. I just realized this myself :) I will delete this repository and restart on a fork.

wschae committed:
Support EnC for lambdas & closures in VB compiler

wschae committed:
Support EnC for lambdas & closures in VB compiler

wschae committed:
Make SynthesizedLambdaMethod implement ISynthesizedMethodBodyImplementationSymbol for EnC

agocke committed:
Replace the native client with the managed one

This deletes the csc2, vbc2, NativeClient, and VBCSC2UI native projects
and replaces the functionality with the managed client implementation in the
build task. At this moment the functionality of the compilers should be
identical to what is currently shipping, but using 100% managed code.

wschae committed:
Support EnC for query expressions in VB compiler

Commit:
Unskip test CS0030ERR_NoExplicitConv02 and remove its dependency on the current culture

wschae committed:
Support EnC for lambdas & closures in VB compiler

- update comments
- update tests

amcasey committed:
Copy ResultProvider sources, rather than binaries

We need to produce two copies of each ResultProvider DLL, one that runs on
NetFX20 and one that is Portable.  We originally believed that we could
create a single DLL that would work for both and so we just copied the
output of each ResultProvider project and signed it with a different
Authenticode certificate.  Now we know that - even though the source of
both versions will be the same - we need to actually recompile for
Portable.

As a first step, we eliminate the copying mechanism and split each
ResultProvider project in two.  To minimize duplication, we introduce
shared projects that hold on to most of the source.

The next step will be to make the two copies reference different
assemblies.

heejaechang committed:
support compilation end action in v1 diagnostic engine

I went thorough v1 diagnostic engine to fully support compilation end actions.

now it should properly cache and report errors to associated services when there is compilation end actions registered.

but admittedly I didn't tighten it as much as I can assuming all these will change once we move to v2. but this should do its job.

pharring commented on Src/Compilers/CSharp/csc2/Csc2.cs:
```diff
@@ -0,0 +1,11 @@
+﻿using Microsoft.CodeAnalysis.BuildTasks;
```
Please add the OSS header to all files.

gafter commented on Src/Compilers/Core/MSBuildTask/BuildClient.cs:
```diff
@@ -48,6 +48,81 @@ private static string GetExpectedServerExeDir()
+            responseTask.Wait();
+
+            int exitCode;
+            var response = responseTask.Result;
```
Can this throw an `AggregateException`?

agocke replied:
It's probably possible. Do you think it's worth trying to put in an exception filter fail fast somewhere here? I don't see a great place to put a `try` block (maybe in TryRunServerCompilation?) -- do you have any suggestions?

gafter commented on src/Compilers/Core/Desktop/CommandLine/CommonCommandLineParser.cs:
```diff
@@ -317,6 +317,45 @@ internal static string RemoveTrailingSpacesAndDots(string path)
+            {
+                const string keepAlive = "/keepalive";
+                var prefixLength = keepAlive.Length;
+                if (arg.StartsWith(keepAlive))
```
Is this intentionally case-sensitive?

msJohnHamby committed:
Diagnostics API changes for registering compilation actions, code block actions, and end actions

This change set introduces the notions of registering compilation actions and code block actions into the diagnostics API. Compilation actions differ from compilation end actions in that they do not have access to per-compilation analyzer state and do not introduce action execution ordering constraints. Code block actions differ from code block end actions in that they do not have access to per-block analyzer state and do not introduce action execution ordering constraints.

This change set also removes RegisterXEndAction methods on contexts other than the corresponding StartXAnalysisContext types. This change is intended to guide analyzer authors towards introducing analyzer state with appropriate lifetime.

Some internal mechanism for the removed registration methods remains. This keeps the mechanism fully general and imposes a very marginal execution cost.

The specific API changes:
 -- Removal of AnalysisContext.RegisterCompilationEndAction. CompilationStartAnalysisContext.RegisterCompilationEndAction remains.
 -- Removal of AnalysisContext.RegisterCodeBlockEndAction and CompilationStartAnalysisContext.RegisterCodeBlockEndAction. CodeBlockStartAnalysisContext.RegisterCodeBlockEndAction remains.
 -- Introduction of AnalysisContext.RegisterCompilationAction.
 -- Introduction of AnalysisContext.RegisterCodeBlockAction and CompilationStartAnalysisContext.RegisterCodeBlockAction.
 -- Renaming of CompilationEndAnalysisContext to CompilationAnalysisContext.
 -- Renaming of CodeBlockEndAnalysisContext to CodeBlockAnalysisContext.

jaredpar committed:
Stop emitting assembly / PDB to a temp directory

When a compilation fails the compiler should not delete or edit the outputs of a previous successful.  They should remain around to satisfy scenarios like F5 on build failure.

Roslyn implemented this logic by doing the following:

- Creating a `Stream` for the assembly / PDB in a temp directory.
- Running the compilation.
- Moving the `Stream` to the desired output location on success.

This introduced a subtle bug into the compiler.  When the source + dest of `File.Move` are on the same volume the destination file will end up with the ACLs it got from being created in the source directory.  In this example it meant the assembly / PDB had the ACLs of the temp directory not the actual output one.

This can produce breaks in cases where the temp and output directory have different inheritable ACLS.  This case comes up for instance in ASP.Net.

To fix this we are going back to the native compiler behavior here.  The creation of the `Stream` objects will be delayed until we are ready to write to disk (happens after the compilation is verified to have no errors).

Because this portion of the compiler is on .NET 4.5.2 the Emit layer cannot open the files directly (no file system API to use).  Instead we pass down the `EmitStreamProvider` abstraction from the desktop layer and let that control the openning of the files.

This change also allows us to remove all of the temp directory logic from the compiler and build protocol.

Note: Whether or not to make this entry point a public API is being considered separately from this thread.  If we decide to make this public it will be done as a separate follow up change.

pharring commented:
LGTM

amcasey committed:
FakeSign the concord binaries

Run FakeSign.exe on the concord binaries that we import as IL and
re-assemble ourselves.  This makes it possible to load them in
CI builds on Jenkins.

CR: @jaredpar; @jasonmalinowski

tmeschter committed:
Update properties when a rule set is removed

Fixes #260.

When a rule set file is removed or renamed we need to update the
related CodeAnalysisRuleSet properties in the project file. Otherwise
the build will break when the compiler complains that it cannot find the
file.

srivatsn committed:
Fixing parentheses simplification in the presence of a conditional member access. Fixes #987

Giving the ConditionalMemberAcces the same precedence as member access while determining associativity fixes the issue.

In VB, removing parentheses simplification works fine. There is a case where cast simplfication doesn't work and I've filed #1025 to track that.

rchande committed:
Merge pull request #1015 from rchande/DebuggerIntellisenseASP

Debugger IntelliSense and linked files

tmat committed:
Merge pull request #1033 from tmat/SEC

SymbolEquivalenceComparer improvements: custom assembly comparer, handli...

tmat committed:
Merge branch 'master' of https://github.com/dotnet/roslyn into LambdaEnC

AdamSpeight2008 committed:
NextAre implementation change.

Using a different implementation, for clarity.

srivatsn committed:
Merge pull request #1026 from srivatsn/inlinetemp

Fixing parentheses simplification in the presence of a conditional member access.

tmat committed:
Merge branch 'master' of https://github.com/dotnet/roslyn into LambdaEnC

tmat committed:
Enable EnC for C# methods containing lambdas and closures

mavasani committed:
Fix execution of CompilationStartActions when only analyzer options for the project change, but the underlying compilation is identical.

Our current assumption for executing CompilationStartActions is that they should be executed exactly once per-compilation per-analyzer. We cache the registered nested actions for every compilation in AnalyzerManager to ensure this. However, if user just changes an additional document for the project, then the analyzer options change, but the underlying compilation is unchanged. Project change causes IDE diagnostic service to clear all cached diagnostics, but when re-computing diagnostics, we end up skipping execution of CompilationStartActions and directly move to nested compilation actions. This can be troublesome in couple of ways. First, we will have duplicate callbacks for all nested actions for the initially invoked compilation start action. Second, and event worse, if the analyzer is reading the contents of the additional documents in the start action and caching them for later use by the nested actions, all the nested actions get stale contents for the additional documents. Apparently, Tom is the first one to go through this agony as the DeclarePublicAPI analyzer that he implemented exactly does the latter.

Fix is to ensure that we execute compilation start actions for every unique compilation start analysis context, i.e. {compilation, analyzer options} tuple for every analyzer.

mavasani committed:
Merge remote-tracking branch 'upstream/master' into Issue530

jaredpar committed:
Fixed a path casing issue

Looks like some of the MSBuild files got checked in under Src instead of
src.  This corrects that.

closes #1045

DustinCampbell committed:
Fix issue introduced by recent change that breaks web scenario

Of course, a ContainedLanguage will never be an IProjectionBuffer. So,
this method always returns an empty enumerable. Instead, the cast should
be against _containedLanguage.DataBuffer.

#hangsheadinshame

tmeschter committed:
Only show check mark if the severities are the same

If the user selects a bunch of diagnostics with differing severities in
Solution Explorer, right-clicks,  and opens the "Set Rule Set Severity"
submenu they will see a check mark next to every applicable severity.
E.g., if one rule is an error and one is a warning, then both Error and
Warn will be checked.

The general feeling is that this will be confusing to users. Instead,
only show a check mark when all the selected rules share the same
severity.

DustinCampbell committed:
Merge pull request #1046 from DustinCampbell/fix-web

Fix issue introduced by recent change that breaks web scenarios

amcasey committed:
Automated checkin: update Concord assemblies to version 22705.00.

tmeschter committed:
Merge pull request #988 from tmeschter/Fix619

Clean up Solution Explorer context menus

KevinH-MS committed:
Make ExpressionCompiler and ResultProvider public...

Reflection APIs used by the dispatcher to create these classes may not be available on all platforms.

brettfo committed:
Merge pull request #1035 from brettfo/localized-resources

fix missed or incorrect localized strings in tests

KevinH-MS committed:
Merge pull request #1049 from KevinH-MS/master

Make ExpressionCompiler and ResultProvider public...

amcasey committed:
Merge pull request #1037 from amcasey/SharedRP

Copy ResultProvider sources, rather than binaries

mavasani committed:
Address CR feedback from Heejae: Avoid unnecessary IEqualityComparer<AnalyzerAndOptions>, set concurrency level for concurrent dictionary, and add early out checks based on reference equality in added Equals overrides.

KevinH-MS committed:
Simplify GetEvaluationResultsAndContinue...

I *think* this should behave the same.  We can remove the range check in the lambda and just call GetEvaluationResultsAndContinue again with "index + 1".  The outer if should catch the index >= numRows case and call the completionRoutine.

tmat committed:
Merge pull request #1040 from tmat/LambdaEnC

Enable EnC for C# methods containing lambdas and closures

cston committed:
Expand root-level synthetic IEnumerable values without intermediate 'Results View'

cston committed:
Merge remote-tracking branch 'upstream/master' into 1114276

KevinH-MS committed:
Merge pull request #1050 from KevinH-MS/master

Simplify GetEvaluationResultsAndContinue...

mavasani committed:
Merge pull request #1044 from mavasani/Issue530

Fix execution of CompilationStartActions when only analyzer options for the project change, but the underlying compilation is identical.

Our current assumption for executing CompilationStartActions is that they should be executed exactly once per-compilation per-analyzer. We cache the registered nested actions for every compilation in AnalyzerManager to ensure this. However, if user just changes an additional document for the project, then the analyzer options change, but the underlying compilation is unchanged. Project change causes IDE diagnostic service to clear all cached diagnostics, but when re-computing diagnostics, we end up skipping execution of CompilationStartActions and directly move to nested compilation actions. This can be troublesome in couple of ways. First, we will have duplicate callbacks for all nested actions for the initially invoked compilation start action. Second, and event worse, if the analyzer is reading the contents of the additional documents in the start action and caching them for later use by the nested actions, all the nested actions get stale contents for the additional documents. Apparently, Tom is the first one to go through this agony as the DeclarePublicAPI analyzer that he implemented exactly does the latter.

Fix is to ensure that we execute compilation start actions for every unique compilation start analysis context, i.e. {compilation, analyzer options} tuple for every analyzer. This change fixes #530

VSadov committed:
Restored requirement for struct constructors to always have formal parameters
Fixes #1029

While overall parameterless constructors in structs are valid from IL perspective, without a convenient way to declare them they were virtually nonexistent. As we performed more and more testing, we kept discovering cases where parameterless struct constructors caused inconsistent behavior in libraries or even in some versions of CLR.

After reconsidering the potential issues arising from breaking long standing assumptions, we decided it was best for our users to restore the requirement on struct constructors to always have formal parameters.

pharring committed:
Remove Logger and RoslynCompilerEventSource

cston committed:
Merge remote-tracking branch 'upstream/master' into 1114276

cston committed:
Incorporate PR feedback

mattwar committed:
Merge pull request #1024 from mattwar/Encoding

Improve Encoding support in SyntaxTrees and Documents

Commit:
Give Roslyn.VisualStudio.Closed.UnitTests IVT to workspaces.dll

Commit:
Move VSTelemetryLoggerTest to Closed

cston committed:
Merge pull request #1051 from cston/1114276

Expand root-level synthetic IEnumerable values without an intermediate "Results View" row

Removing the intermediate "Results View" row for synthetic values allows LINQ queries to be expanded automatically in the Immediate Window.

tmeschter committed:
Disable the UnusedDeclarationsAnalyzer by default

Related to issue #1058.

The analyzer reports thousands of spurious errors when running over
Roslyn.sln. In order to move forward to newer builds of the toolset
compilers and diagnostics, it needs to be disabled.

mattwar committed:
Add ImportAdder API

mattwar committed:
Merge remote-tracking branch 'upstream/master' into ImportAdder

mavasani committed:
Fixes #950: Seal public methods of DiagnosticAnalyzer that can be overridden and can throw: Equals,GetHashCode and ToString. This is required as we use analyzers as keys for various caches in our diagnostic engines.

Also fix an issue pointed out by CyrusN: Don't report analyzer diagnostics for OperationCancelledException during analyzer execution.

tmeschter committed:
Merge pull request #1059 from tmeschter/TurnOffUnusedDeclarationAnalyzer

Disable the UnusedDeclarationsAnalyzer by default

AdamSpeight2008 committed:
Clearing Up additional blank line.

Also tidy up character sequences.

mavasani committed:
Merge remote-tracking branch 'upstream/master' into Issue950

Commit:
Merge pull request #1047 from heejaechang/CompilationEnd

support compilation end action in v1 diagnostic engine

amcasey committed:
Merge pull request #1019 from amcasey/GH1002

FakeSign the concord binaries

cston committed:
Merge remote-tracking branch 'upstream/master' into 1115030

tmeschter committed:
Use diagnostic property bag in DeclarePublicAPI

Use the new Diagnostic property bag to pass information from the
DeclarePublicAPI analyzer to the DeclarePublicAPI fix. Specifically, we
pass the symbol name so the fix doesn't have to recompute it.

rchande committed:
Merge pull request #1056 from rchande/CloseVSTest

Move VSTelemetryLoggerTests to a closed source test project

pharring committed:
Merge pull request #1057 from pharring/RemoveLogger

Remove Logger and RoslynCompilerEventSource

nguerrera committed:
Upgrade to RC versions of S.C.Immutable and S.R.Metadata

nguerrera committed:
React to breaking changes in S.C.Immutable RC

jmarolf committed:
Add imports for extension add methods

Offer to add imports\namespaces for extension methods that apply to
collection initializers.

Fixes #269

gafter committed:
Add a doc about the change in unicode version.

gafter committed:
Merge pull request #1070 from gafter/docs

Add a doc about the change in unicode version.

nguerrera committed:
Set expected S.N.Vectors projected version to 4.0.0.0

closes #1077

pharring committed:
Add typing training scenario for ASP.Net projects.

tmat committed:
Enable hosts to compile strong-named assemblies on all platforms by supplying public key via compilation options and also enable OSS signing with such key

tmat committed:
Merge branch 'CryptoPublicKey' of https://github.com/tmat/roslyn into CryptoPublicKey

tmat committed:
Updates the ENC closure CDI to distinguish between static and this-only closures

tmat committed:
Merge pull request #1071 from tmat/CryptoPublicKey

Enable hosts to compile strong-named assemblies on all platforms by supp...

Commit:
Rename of VB Property Backing Fields

Fixes #905

Add rename support for VB property backing fields. We now correctly
exclude the leading '_' from the rename process and allow rename to be
invoked from references to the backing field.

pharring committed:
Merge pull request #1001 from AdamSpeight2008/(NextAre)

Implemented NextAre

cston committed:
Use actual value and type for ', results' - feedback from PR#1051

mavasani committed:
Merge pull request #1062 from mavasani/Issue950

Fixes #950 : Seal public methods of DiagnosticAnalyzer that can be overridden and can throw: Equals,GetHashCode and ToString. This is required as we use analyzers as keys for various caches in our diagnostic engines.

Also fix an issue pointed out by CyrusN: Don't report analyzer diagnostics for OperationCanceledException during analyzer execution.

mavasani committed:
Merge remote-tracking branch 'upstream/master' into Issue950

shyamnamboodiripad committed:
Delete the public TriggerDiagnosticDescriptor type.

And seal its base type DiagnosticDescriptor.

(Fixes #789)

msJohnHamby committed:
Merge remote-tracking branch 'upstream/master' into CompilationAction

mavasani committed:
Fixes #1025 : Replace '!=' check for comparing symbols in AbstractSpeculationAnalyzer with SymbolsAreCompatible check, which basically uses Equals method for comparison. This was causing two equivalent symbols across semantic model and it's speculative model to be reported as incompatible, causing all simplifications to bail out for ?. expression in VB.

msJohnHamby committed:
Merge pull request #1043 from JohnHamby/CompilationAction

Diagnostics API changes for registering compilation actions, code block actions, and end actions

msJohnHamby committed:
Corrected the doc comment for CodeBlockAnalysisContext

msJohnHamby committed:
Merge pull request #1084 from JohnHamby/CorrectComments

Corrected the doc comment for CodeBlockAnalysisContext

heejaechang committed:
Add support for compilation end code fix.

now code fix should support diagnostics from compilation end action.

a few options are added to test experience which can be changed from Tools/Options/Roslyn/Diagnostic pane.

pharring committed:
Merge pull request #1072 from pharring/VenusPgoTyping

Add typing training scenario for ASP.Net projects.

jaredpar committed:
Respond to PR feedback

This change takes the following steps:

- Changes EmitStreamProvider to wrap a single Stream instead of a group
of Stream.
- Changes the emit behavior to consider Analyzer errors before
attempting to write the PE / PDB Stream to disk.

cston committed:
Extract string constant

cston committed:
Merge pull request #1080 from cston/1114276

Use actual value and type for ', results'

[Feedback from PR#1051]

cston committed:
Add state machine local for catch block exception

cston committed:
Merge remote-tracking branch 'upstream/master' into 1115030

Commit:
Test for property backing field rename conflicts

jaredpar committed:
Fix a build break

There appears to be a merge issue with the NuGet upgrade change that
caused a build break.  This fixes that.

mavasani committed:
Address CR feedback: Add the github issue link as the WorkItemAttribute description for fixed test.

brettfo committed:
use localized resources in unit tests

basoundr committed:
Getting Parent of SyntaxNodeOrToken does not need conversion to Node or Token.

Fix internal TFS bug 1130555 : Getting Parent of SyntaxNodeOrToken
does not need conversion to Node or Token.

When converting a SyntaxNodeOrToken to a Node or Token use IsNode or IsToken.

In this particular instance, we have introduced the check in
InlIneTemporaryCodeRefactoringProvider to prevent crash. We will still
have to find the scenario that triggered this situation.

mavasani committed:
Merge pull request #1082 from mavasani/Issue1025

Fixes #1025 : Replace '!=' check for comparing symbols in AbstractSpeculationAnalyzer with SymbolsAreCompatible check, which basically uses Equals method for comparison. This was causing two equivalent symbols across semantic model and it's speculative model to be reported as incompatible, causing all simplifications to bail out for ?. expression in VB.

brettfo committed:
Merge pull request #1088 from brettfo/localized-resources

use localized resources in unit tests

amcasey committed:
Hide non-file nodes in Concord.csproj.

tmeschter committed:
Add a CI build script

Instances of vbcscompiler.exe may hang around for a few minutes after a
Jenkins run, locking files and causing new runs on the same system to
fail. To address this, we add a script to setup the environment, run the
build and test, and then kill any instances of vbcscompiler.exe that are
left around.

amcasey committed:
Manually update IL of Microsoft.VisualStudio.Debugger.Metadata.

amcasey committed:
Update ResultProvider with breaking LMR changes

Microsoft.VisualStudio.Debugger.Metadata is going to be making some
breaking changes.  These are the corresponding ResultProvider changes.

tmat committed:
Merge pull request #1076 from tmat/ClosureKindInPDB

Updates the ENC closure CDI to distinguish between static and this-only closures

tmat committed:
Use empty array for unspecified public key instead of default(ImmutableArray<byte>) to avoid issues with JSON serialization

dpoeschl committed:
Merge pull request #1078 from dpoeschl/RenameVBPropertyBackingField

Rename of VB Property Backing Fields

shyamnamboodiripad committed:
Merge pull request #1065 from shyamnamboodiripad/DeleteTriggerDiagnosticDescriptor

Delete the public TriggerDiagnosticDescriptor type.

And seal its base type DiagnosticDescriptor.

(Fixes #789, #843)

gafter committed:
Merge pull request #924 from gafter/fix803

Make the order of symbols returned from DataFlowAnalysis deterministic

jaredpar committed:
Merge pull request #1090 from tmeschter/AddCIBuildCmd

Add a CI build script

jmarolf committed:
Moving VB tests to correct class.

heejaechang committed:
fix crash due to control following through next statement

Commit:
Merge pull request #1095 from heejaechang/FixCrashPersist

fix crash due to control following through next statement

jaredpar committed:
Revert "Set expected S.N.Vectors projected version to 4.0.0.0"

This reverts commit b30cf91c76b0f49fa1e3dd5ea472b47a9972737a.

jaredpar committed:
Revert "React to breaking changes in S.C.Immutable RC"

This reverts commit 902153465fc22ce430dcb049a44c7bb528c527b4.

jaredpar committed:
Revert "Upgrade to RC versions of S.C.Immutable and S.R.Metadata"

This reverts commit b37052d0a818354f994f3908c7184a9c62189deb.

mattwar committed:
Add GenerationOptionsProvider

Commit:
No naked argument Exceptions

Update argumentexceptions to include meaningful information.

mattwar committed:
Change ImportAdder to pure static class

cston committed:
Add hoisted catch exception to the local scope

plnelson committed:
Implement support for inspecting native COM objects in the new EE.  The native view of the COM object now shows up under a "Native View" node.

Commit:
Switch to nameof and template strings.

Commit:
Provide explicit StringComparer.Ordinal to methods in Assert type

amcasey committed:
Automated checkin: update Concord assemblies to version 22706.00.

tmat committed:
Merge pull request #1091 from tmat/master

Use empty array for unspecified public key instead of default(ImmutableArray<byte>) ...

jmarolf committed:
Responding to code review feedback

using SpecializedTasks to avoid allocation.

cston committed:
Merge pull request #1102 from cston/1115030

Add hoisted catch block exception to the local scope so the local is available to the EE

jmarolf committed:
Merge pull request #1069 from jmarolf/ImportsForAddExtensionMethods

Add imports for extension add methods

tmeschter committed:
Catch exceptions from user code

The Solution Explorer calls `DiagnosticAnalyzer.SupportedDiagnostics` to
determine the list of items to show. This property is implemented by a
3rd-party (the analyzer author) and could potentially throw.

Here we move to using the `DiagnosticAnalyzerService` to obtain this
data; it will catch any exceptions and handle reporting them to the user
as appropriate.

Fixes #898.

tmeschter committed:
Skip a couple of failing unit tests

These failures are tracked by #1109.

pharring committed:
Introduce AbstractLookupSymbolsInfo.Count property and use it.

VSadov committed:
Merge pull request #1106 from VSadov/parameterlessVB

Restored requirement for struct constructors to always have formal param...

Commit:
Increment BloomFilter serialization version

When a BloomFilter is serialized, it includes a serialization format
version number, some other metadata, and a bitarray. Prior to PR #877,
the bitarray portion also included its own serialization format version
number, so the data serialized was Format 1 of the BloomFilter
containing Format 0 of the bitarray. PR #877 removed the (redundant)
serialization format version number from the bitarray portion but did
not also version the BloomFilter's serialization format, causing the
previous bitarray version number (0) to be read as part of the bitarray
itself. We believe this would sometimes lead to extremely long Find
References calls as the BloomFilter continually tried and failed to
understand the data in the bitarray.

dpoeschl committed:
Merge pull request #1110 from tmeschter/DisableUnitTests

Skip a couple of failing unit tests

pharring committed:
Merge pull request #1112 from pharring/LookupSymbol

Introduce AbstractLookupSymbolsInfo.Count property and use it.

amcasey committed:
Add regression tests for DevDiv #1134746.

dpoeschl committed:
Merge pull request #1113 from dpoeschl/RevBloomFilterSerializationFormat

Increment BloomFilter serialization version

gafter committed:
Allow VB Shared New to initialize a Shared field of another instance.
Fixes #1028
Closes #1116

amcasey committed:
Improve EvaluationContext cache invalidation

Ignoring references, which are obviously the high-order bit, the old
strategy was to confirm that the methodToken, methodVersion, and smallest
containing scope were the same as the current context.  This had two
problems:

1) C# hoisted locals are not reflected in scopes and so we were not
invalidating the cache when we stepped across scope boundaries in async
and iterator methods.

2) We had to access the PDB in order to check whether the cache was still
valid (to figure out the smallest containing scope).

We propose a new approach: compute the largest span around the current IL
offset in which there are no scope boundaries (including hoisted local
scopes).  Then, when check cache validity by doing a simple span
containment check (plus methodToken and methodVersion, obviously).

Downside 1: We need all scopes in the method, not just the containing
ones.

Downside 2: If you somehow step around a nested scope (e.g. using
breakpoints), the new approach will invalidate the cache, whereas the old
one did not.

Commit:
Merge pull request #1111 from pnelsonmsft/cominspection

EE: Add support for COM object Inspection

wschae committed:
Support EnC for lambdas & closures in VB compiler

- Resolve Code review feedback

wschae committed:
Use LambdaDebugInfo constant

wschae committed:
Merge pull request #655 from wschae/VBLambda

Support EnC for lambdas & closures in VB compiler

stephentoub committed:
Disable Array.Empty params array optimization in expression lambdas

Commit:
Rename "Src" to "src", so all the files line up.

wschae committed:
Adjust sequence points in VB let clause

VB compiler does not correctly place sequence points in let clause. As a result the debugger does not hit a sequence point set to let clause. Before this fix, we have the following in PDB:

	<sequencePoints>
		<entry offset="0x0" hidden="true" document="0"/>
		<entry offset="0x1" startLine="8" startColumn="25" endLine="8" endColumn="30" document="0"/>
	</sequencePoints>

After having this fix, we have as in Dev12:

            <sequencePoints>
                <entry offset="0x0" startLine="8" startColumn="25" endLine="8" endColumn="30" document="0"/>
            </sequencePoints>

pharring committed:
Use a custom encoding for TextSpans coming from the StringBreaker used in PatternMatcher.

pharring committed:
Adjusted bit encoding to allow for more spans. Review feedback. Refactored creation routines.

wschae committed:
Calculate syntax offsets of lambdas defined in a partial class

wschae committed:
Merge pull request #1124 from wschae/perf

Calculate syntax offsets of lambdas defined in a partial class

stephentoub committed:
Fix typo in comment

stephentoub committed:
Change hashing light-up to first consider standalone contracts

Until Roslyn is changed to statically reference the System.Security.Cryptography.* contract assemblies, its reflection support for finding HashAlgorithms needs to be updated to first constder the standalone implementations; mscorlib will not always contain these types.

jaredpar committed:
Match native compiler PDB behavior

This causes the compiler to match the native one in terms of PDB
behavior.  It will only pass along the Stream value to Initialize in the
cases where a Stream is explicitly provided.  In the other cases,
basically the raw command line, it will pass along only the file name.
This puts the responsibility of creating / writing the file solely in
ISymUnmanagedWriter as it previously was via WritePdbToOutput

jaredpar committed:
Unify Stream passing API

This unifies the Stream passing API to prefer EmitStreamProvider over
Stream in the Emit APIs.  Even though we are not currently taking
advantage of the delay creation mechanism for the command line compiler
there was a desire to have consistent parameter values where possible.
The advantage is it lets us switch to making XML documentation delay
created as well with a very small change in the future.

jaredpar committed:
Delay PE creation till last possible moment

This change further delays the creation of the PE Stream until the
absolute last possible moment.

jaredpar committed:
Merge pull request #1100 from paulvanbrenk/master

No naked argument Exceptions

Commit:
Merge pull request #1115 from VladimirReshetnikov/bug1119616

Fix #1108. Provide explicit `StringComparison.Ordinal` in invocations of Xunit methods.

pharring committed:
Merge pull request #1122 from pharring/StringBreaker

PERF: Reduce List<TextSpan> allocations in PatternMatcher

jaredpar committed:
Fix cibuild.cmd

Make sure the error checking is done properly.

- msbuild needs to be checked for failure
- taskkill failure shouldn't matter

jaredpar committed:
Kill server on build failure too.

jaredpar committed:
Ensure a file name for PDB

The ISymUnmanagedWriter2 interface does not require a file name be
passed.  It can successful emit a PDB without it.  The only method which
actually requires it is GetDebugInfo.

The Compilation type will call this member frequently during PDB
creation.  Hence if we have a Stream only we need to manufacture a name
to use for it.

jaredpar committed:
Merge pull request #1120 from beannaich/fix-src-folders

Rename "Src" to "src", so all the files line up.

tmeschter committed:
Merge pull request #1048 from tmeschter/Fix260

Update properties when a rule set is removed

tmeschter committed:
Merge pull request #1087 from tmeschter/UpdateDeclareAPIFix

Use diagnostic property bag in DeclarePublicAPI

DustinCampbell committed:
Tweak Find All References display to reduce some noise

* Display logical path rather than full file path. I.e.
<ProjectName>\<Folders>\<FileName>
* Use gray text to indicate definitions that have no references rather
than definitions defined in metadata.
* Show "(0 references)" when there aren't any found.
* Don't fully-qualify definitions.

amcasey committed:
Disable the ResultProvider tests in Jenkins until the new Microsoft.VisualStudio.Debugger.Metadata.dll is publicly available.

wschae committed:
Merge pull request #1121 from wschae/LetClause

Adjust sequence points in VB let clause

balajikris committed:
Properly handle conditional access expressions in CanReplaceWithLValue
extension method

DustinCampbell committed:
Merge pull request #1133 from DustinCampbell/tweak-far-display

Tweak Find All References display to reduce some noise

mattwar committed:
Satisfy Nikov

gafter committed:
Merge branch 'fix1028'

mattwar committed:
Use diagnostics constants for clarity

amcasey committed:
Address PR feedback.

KevinH-MS committed:
Merge pull request #1105 from amcasey/PortableLmr

Update ResultProvider with breaking LMR changes

mattwar committed:
Merge pull request #1063 from mattwar/ImportAdder

Add ImportAdder API

stephentoub committed:
Merge pull request #1117 from stephentoub/arrayempty_exprlambda

Disable Array.Empty params array optimization in expression lambdas

mattscheffer committed:
Merge pull request #1 from dotnet/master

updating personal fork to latest commit

mattscheffer committed:
Clean up c++ projects

Commit:
Remove pending notifications

If we do not remove pending notifications when we cancel them, we may
try to cancel them twice.

jaredpar committed:
Run code formatter on VB compiler source code

This is the initial run of the code formatter on the VB compiler source code.  Further runs I'll expand on this to include all VB code in a future run.

mattscheffer committed:
Removing auto generated redundant line

Commit:
Merge pull request #922 from VladimirReshetnikov/fixCulture

Unskip and fix test CS0030ERR_NoExplicitConv02

pharring committed:
Optimize AccumulateTextChanges for the simple case of typing on a line

pharring committed:
Changed the default compiler server keepalive to 10 minutes (was 5 minutes)

pharring committed:
Merge pull request #1143 from pharring/KeepAlive10

Changed the default compiler server keepalive to 10 minutes

balajikris committed:
Make GenerateVariable work inside conditional access expressions.

tmeschter committed:
Disable some broken unit tests.

tmeschter committed:
Merge pull request #1147 from tmeschter/DisableUnitTests

Disable some broken unit tests.

pharring committed:
Made MetadataDecoder.GetSignatureCountsOrThrow static to reduce allocations in PEMethodSymbol.Arity

rchande committed:
Merge pull request #1138 from rchande/ProjectReloadPerf

Remove pending notifications from SolutionEventMonitor

Fixes #1136

stephentoub committed:
Add try/catch around Type.GetType

pharring committed:
Merge pull request #1142 from pharring/AccumulatedTextChanges

Optimize AccumulateTextChanges for the simple case of typing on a line

mattwar committed:
Change CodeModel to not auto-simplify added imports

heejaechang committed:
Merge branch 'master' of https://github.com/dotnet/roslyn into CompilationEnd

stephentoub committed:
Merge pull request #1127 from stephentoub/hashing_lightup

Change hashing light-up to first consider standalone contracts

amcasey committed:
Merge pull request #1150 from mattwar/CodeModelAddImportFix

Change CodeModel to not auto-simplify added imports

gafter committed:
Detect and report attempt to use abstract event through base
Fixes #406

heejaechang committed:
move some of code to new C# language feature and comment fix up

heejaechang committed:
fix invalid assert and more spelling fix

heejaechang committed:
Merge branch 'master' of https://github.com/dotnet/roslyn into CompilationEnd

pharring committed:
Merge pull request #1148 from pharring/PEMethodSymbolArity

PERF: MetadataDecoder allocations for PEMethodSymbol.Arity

heejaechang committed:
Merge branch 'master' of https://github.com/dotnet/roslyn into TextVersion

heejaechang committed:
re-ordered signature

made signature consistent with overloads.

Commit:
Merge pull request #1132 from heejaechang/CompilationEnd2

Add support for compilation end code fix.

mavasani committed:
Slight tweaks in handling of OperationCanceledExceptions during analyzer execution:

1. Don't swallow OperationCanceledException fired with our cancellelationToken. These are supposed to be handled up the stack and swallowing them would mean unnecessary computation would be done for result that would eventually be ignored.
2. Analyzer callbacks that register actions need custom handling of all OperationCanceledExceptions in AnalyzerManager. Multiple threads trying to fetch registered analyzer actions might share the same task for computing registered actions, and if cancellation is fired on the source of this task, we want to re-attempt invoking the register callbacks, otherwise the analyzer will never execute.

mavasani committed:
Merge remote-tracking branch 'upstream/master' into AnalyzerExecutionCancellation

heejaechang committed:
don't handle code link ourselves but let error list handle it

Commit:
Merge pull request #1159 from heejaechang/SignatureChange

re-ordered signature

mavasani committed:
Don't swallow OperationCanceledException fired with our cancellelationToken. These are supposed to be handled up the stack and swallowing them would mean unnecessary computation would be done for result that would eventually be ignored.

mavasani committed:
Merge pull request #1160 from mavasani/AnalyzerExecutionCancellation

Don't swallow OperationCanceledException fired with our cancellelationToken. These are supposed to be handled up the stack and swallowing them would mean unnecessary computation would be done for result that would eventually be ignored.

tmeschter committed:
Add a method to `IDiagnosticAnalyzerService`

Add the `GetDiagnosticDescriptors(DiagnosticAnalyzer)` method to the
`IDiagnosticAnalyzerService` interface.

basoundr committed:
Dont disturb whitespace preceding comments

Fix #1151 : Dont disturb the whitespaces preceding the comments which
are also not part of indentation to be converted to tabs if 'Keep Tabs'
option is set.

tmeschter committed:
Merge pull request #1139 from tmeschter/HardenSolutionExplorer

Catch exceptions from user code

Commit:
Merge pull request #1161 from heejaechang/F1Help

don't handle code link ourselves but let error list handle it

tmat committed:
Implement EnC support for range variables in C# queries, improve query clause matching

tmat committed:
Merge pull request #1168 from tmat/C#Q

Implement EnC support for range variables in C# queries, improve query clause matching

amcasey committed:
Remove my alias from all TODOs.

gafter committed:
Fix a build break by adding a parameter to an otherwise parameterless struct ctor

gafter committed:
Merge pull request #1170 from gafter/buildbreak

Fix a build break by adding a parameter to an otherwise parameterless struct ctor

balajikris committed:
Merge pull request #1149 from balajikris/ExtractMethod

Fix Extract Method and Generate Variables in Conditional Access Expressions.

Fixes #751.

Properly handle ConditionalAccessExpressions in ExpressionSyntaxExtensions::CanReplaceWithLValue. When conditional access expressions begin with some form of member binding expression (starts with the dot operator), they cannot be replaced with an LValue. @jmarolf contributed commit cf3dd54

This issue does not affect VB as VB language parses this differently than C#.

Pilchie committed:
When validating references, ignore duplicates

It seems that VsProject's references may include duplicate versions of
mscorlib for portable projects, so remove duplicates before comparing
against our workspace set of references.

This fixes #1083.

CyrusNajmabadi committed:
Introduce a new AbstractEncProject for CSharpProject and VisualBasicProject to extend.

This project does the appropriate Enc hookup, and implements the interfaces that the debugger
QIs for to do Enc interaction.

For C# and VB, nothing will change.  For TypeScript, we will continue subclassing AbstractProject.
By doing this, we will not interact with the Enc system at all.  This will help avoid undesirable
behavior (like having our files become read-only when entering debugging).

basoundr committed:
Merge pull request #1089 from basoundr/fix1130555NREInlineTemp

Check SyntaxNodeOrToken is a SyntaxNode before converting

DustinCampbell committed:
Update Code Model to support ReadOnly and WriteOnly VB auto-properties

VB 14 supports ReadOnly and WriteOnly auto-properties. Code Model's
CodeProperty2.ReadWrite should return the correct value for them.

basoundr committed:
Remove UseTabOnlyForIndentation and Fix broken testcases

Remove UseTabOnlyForIndentation since this is a public option & there
are no testcases for this option. With the changes the option is no
longer used making it obsolete. Fix the broken testcase which was
asserting the undesired behavior

mattscheffer committed:
Merge pull request #1137 from mattscheffer/master

Cleaned up C++ project targets

jaredpar committed:
Fix up misaligned comments

This fixes up the misaligned comments that came from the formatting bug
described in #1177.

closes #1140

amcasey committed:
Merge pull request #1153 from amcasey/DD1134746

New EvaluationContext cache-invalidation strategy

CyrusNajmabadi committed:
Only mark buffers readonly if they're part of a project that supports Enc.

tmat committed:
Enable editing methods with anonymous types and remove unused rude edit kinds

CyrusNajmabadi committed:
Sort usings.

CyrusNajmabadi committed:
CR feedback.

tmat committed:
Merge pull request #1179 from tmat/master

Enable editing methods with anonymous types and remove unused rude edit kinds

CyrusNajmabadi committed:
Merge pull request #1172 from CyrusNajmabadi/encProject

Introduce a new AbstractEncProject for CSharpProject and VisualBasicProject to extend.

gafter committed:
Merge pull request #1152 from gafter/fix406

Detect and report attempt to use abstract event through base

jaredpar committed:
Fix a Release build break

There were uses of private fields in #if blocks that needed to be
adjusted to the renamed fields.

gafter committed:
Fix a NRE in VB query binding in an error scenario
Fixes #1099

gafter committed:
Merge branch 'master' of https://github.com/dotnet/roslyn into fix1099

gafter committed:
Merge pull request #1182 from gafter/fix1099

Fix a NRE in VB query expression binding

VSadov committed:
Made use of extension Add methods in Expression lambdas a compiler error.

Expression Trees specifically require that element initializer methods are instance methods.
Fixes #310

Pilchie committed:
Merge pull request #1173 from Pilchie/Fix1083

When validating references, ignore duplicates

VSadov committed:
Replaced the error message for invalid use of extension Add method in expression lambda.

basoundr committed:
Merge pull request #1166 from basoundr/fix1151WhitespaceBeforeComment

Dont disturb whitespace preceding comments

KevinH-MS committed:
Fallback matching for manifest "AssemblyFiles"...

In some scenarios, it appears that the strings for AssemblyFiles contain file extensions (.dll, etc).  However, the module names in the "modulesByName" (ModuleMetadata.Name) will not.  To work around this, implement a "fallback" match that removes the possible file extension.

basoundr committed:
Dont Assume Certain Node Are Always Parented

Fix 1118 : Dont assume that some of the nodes are always parented.
Always check if the parent is not null before accessing the parent.

amcasey committed:
Automated checkin: update Concord assemblies to version 22711.00.

basoundr committed:
Merge pull request #1188 from basoundr/fix1118FormatCrashOnBlock

Dont Assume Certain Node Are Always Parented

DustinCampbell committed:
Convert tabs to spaces

VSadov committed:
Merge pull request #1183 from VSadov/AddFix

Made use of extension Add methods in Expression lambdas a compiler error...

tmeschter committed:
Disable a few unit tests

These are failing in certain build configurations.

tmeschter committed:
Merge pull request #1191 from tmeschter/DisableUnitTests

Disable a few unit tests

DustinCampbell committed:
Merge pull request #1174 from DustinCampbell/fix-issue-851

Update Code Model to support ReadOnly and WriteOnly VB auto-properties

KevinH-MS committed:
Merge pull request #1186 from KevinH-MS/master

Fallback matching for manifest "AssemblyFiles"...
